File: librarymanager/librarymanager-be/src/main/java/ru/elenacherev/librarymanager/mapper/OrderMapper.java
package ru.elenacherev.librarymanager.mapper;

import ru.elenacherev.librarymanager.api.dto.Order;
import ru.elenacherev.librarymanager.domain.entity.EditionEntity;
import ru.elenacherev.librarymanager.domain.entity.OrderEntity;
import ru.elenacherev.librarymanager.domain.entity.ReaderEntity;

public class OrderMapper {
    public static Order map(OrderEntity entity) {
        return Order.builder()
                .orderId(entity.getOrderId())
                .completeDate(entity.getCompleteDate())
                .orderDate(entity.getCompleteDate())
                .orderState(entity.getOrderState())
                .readerFio(entity.getReader()!=null
                		?entity.getReader().getFio()
                		:""
                )
                .editionTitle(entity.getEdition()!=null
                		?entity.getEdition().getTitle()
                		:""
                )
                .readerId(entity.getReader()!=null
                		?entity.getReader().getReaderId()
                		:0
                )
                .editionId(entity.getEdition()!=null
                		?entity.getEdition().getEditionId()
                		:0
                )
                .build();
    }

    public static OrderEntity map(
    		Order dto, 
    		ReaderEntity readerEntity,
    		EditionEntity editionEntity
    ) {
        return map(dto, new OrderEntity(), readerEntity, editionEntity);
    }

    public static OrderEntity map(
    		Order dto, 
    		OrderEntity entity,
    		ReaderEntity readerEntity,
    		EditionEntity editionEntity
    ) {
        entity.setCompleteDate(dto.getCompleteDate());
        entity.setOrderDate(dto.getOrderDate());
        entity.setOrderState(dto.getOrderState());
        entity.setReader(readerEntity);
        entity.setEdition(editionEntity);
        return entity;
    }
}

File: librarymanager-be/src/main/java/ru/elenacherev/librarymanager/mapper/OrderMapper.kt
package ru.elenacherev.librarymanager.mapper

import ru.elenacherev.librarymanager.api.dto.Order
import ru.elenacherev.librarymanager.domain.entity.EditionEntity
import ru.elenacherev.librarymanager.domain.entity.OrderEntity
import ru.elenacherev.librarymanager.domain.entity.ReaderEntity

fun OrderEntity.map() = Order(
    orderId = orderId,
    completeDate = completeDate,
    orderDate = orderDate,
    orderState = orderState,
    readerFio = reader.fio,
    editionTitle = edition.title,
    readerId = reader.readerId ?: 0,
    editionId = edition.editionId ?: 0
)

fun Order.map(readerEntity: ReaderEntity,
              editionEntity: EditionEntity
) = OrderEntity(
    completeDate = completeDate,
    orderDate = orderDate,
    orderState = orderState,
    reader = readerEntity,
    edition = editionEntity
)

fun Order.map(
    entity:OrderEntity,
    readerEntity: ReaderEntity,
    editionEntity: EditionEntity
) = entity.also {
    it.completeDate = completeDate
    it.orderDate = orderDate
    it.orderState = orderState
    it.reader = readerEntity
    it.edition = editionEntity
}